### PR TITLE
Split the service initialisation up into seperate functions

### PR DESCRIPTION
**Pull request by expenses**
This is an attempt at what @bkchr talks about here: https://github.com/paritytech/substrate/pull/6309#issuecomment-641911216. This makes the `ServiceBuilder::build_common` a lot simpler, and will make it (in the long run) a lot easier to get rid of.

### CODE REVIEW

**Comment by cecton** [comment]
tbh I thought @bkchr wanted the refactoring to be in one go, I didn't expect an intermediary PR?

**Comment by expenses** [comment]
@cecton Where possible I'd really like to break this task up into pieces. It's a lot easier to review and merge in.

**Comment by cecton** [comment]
I do agree

File: client/service/src/builder.rs [hunk 1]
```diff
@@ -1495,3 +1241,384 @@ ServiceBuilder<
...
+		telemetry!(
+			SUBSTRATE_INFO;
+			"system.network_state";
+			"state" => network_state,
+		);
+		ready(())
+	}).await;
+}
+
+async fn informant<TBl, TBackend, TExec, TRtApi, TExPool>(
```
Comment (bkchr) [inline client/service/src/builder.rs:1372]: What is the reason of this function? It just wraps another function call?

File: client/service/src/builder.rs [hunk 2]
```diff
@@ -1495,3 +1241,384 @@ ServiceBuilder<
...
+	let finality_stream = client.finality_notification_stream()
+		.map(Into::into)
+		.fuse();
+
+	futures::stream::select(import_stream, finality_stream)
+		.for_each(|evt| txpool.maintain(evt))
+		.await
+}
+
+async fn offchain_notifications<TBl, TBackend, TExec, TRtApi>(
```
Comment (bkchr) [inline client/service/src/builder.rs:1266]: I would have assumed that this is moved to the offchain crate.

File: client/service/src/builder.rs [hunk 3]
```diff
@@ -1495,3 +1241,384 @@ ServiceBuilder<
 		self.build_common()
 	}
 }
+
+// Inform the tx pool about imported and finalized blocks.
+async fn txpool_notifications<TBl, TBackend, TExec, TRtApi, TExPool>(
```
Comment (bkchr) [inline client/service/src/builder.rs:1246]: And this to the tx-pool crate.

**Comment by bkchr** [comment]
> tbh I thought @bkchr wanted the refactoring to be in one go, I didn't expect an intermediary PR?

No, I just wanted to create a proper implementation, which the last one wasn't. 

This one begins with splitting the stuff, to prepare a better implementation where we can drop the builder and probably the service.

File: client/service/src/builder.rs (hunk 1 above)
Comment (expenses) [reply client/service/src/builder.rs:1372]: it was to simplify the function call, but I think I can just move those changes into `sc_informant::build` itself.

File: client/service/src/builder.rs (hunk 2 above)
Comment (expenses) [reply client/service/src/builder.rs:1266]: We'd need to move `Client` out of `sc-service` again in order to do that. @seunlanlege what are your thoughts on this?

Comment (expenses) [reply client/service/src/builder.rs:1266]: Wait, no, we can just use the client api :facepalm:

**Pull request merged**